File: benchmark/c/cpw/maximize_volume_cuboid_given_sum_sides/maximize_volume_cuboid_given_sum_sides.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int maximize_volume_cuboid_given_sum_sides ( int s ) ;
int maximize_volume_cuboid_given_sum_sides ( int s ) {
  int maxvalue = 0;
  for ( int i = 1;
  i <= s - 2;
  i ++ ) {
    for ( int j = 1;
    j <= s - 1;
    j ++ ) {
      int k = s - i - j;
      maxvalue = max ( maxvalue, i * j * k );
    }
  }
  return maxvalue;
}

File: translation_result/else_bms/fail/c/cpw/maximize_volume_cuboid_given_sum_sides/gemini/Restart/maximize_volume_cuboid_given_sum_sides.rs

use std::cmp;

fn maximize_volume_cuboid_given_sum_sides(s: i32) -> i32 {
    let mut maxvalue = 0;
    for i in 1..=s - 2 {
        for j in 1..=s - 1 {
            let k = s - i - j;
            maxvalue = cmp::max(maxvalue, i * j * k);
        }
    }
    maxvalue
}
